tests: drivers: flash: common: multiple failures on disable_spi_nor test configuration

### Describe the bug

- **What target platform(s) are you using?**
  - [black_f407ve/stm32f407xx](https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729502)
  - [adafruit_feather_stm32f405/stm32f405xx](https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729501)
  - [gd32f450i_eval/gd32f450](https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729559)
  - [gd32f450z_eval/gd32f450](https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729570)
  - [legend@35/stm32f070xb](https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729577)
  - [mini_stm32h743/stm32h743xx](https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729651)
  - [pinetime_devkit0/nrf52832](https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729716)
  - [rv32m1_vega/openisa_rv32m1/zero_riscy](https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729715)
- **What have you tried to diagnose and workaround this issue?** disable the failing test until the test or platforms are reworked to pass
- **Is this a regression? If yes, have you been able to "git bisect" it to a specific commit?** Probably? I can't see the test working for arbitrary platforms since it was added in d8be41c00a1ca2dcdc97031436dc851a6f5137b9

### Regression

- [ ] This is a regression.

### Steps to reproduce

**Steps to reproduce the behaviour**
1. `west twister --build-only -c -p black_f407ve/stm32f407xx -p adafruit_feather_stm32f405/stm32f405xx -p gd32f450i_eval/gd32f450 -p gd32f450z_eval/gd32f450 -p legend@35/stm32f070xb -p mini_stm32h743/stm32h743xx -p pinetime_devkit0/nrf52832 -p rv32m1_vega/openisa_rv32m1/zero_riscy -T tests/drivers/flash/common -s drivers.flash.common.disable_spi_nor`
2. See errors

### Relevant log output

```shell
/home/cfriedt/zephyrproject/zephyr/include/zephyr/device.h:96:41: error: '__device_dts_ord_DT_N_NODELABEL_storage_partition_PARENT_PARENT_ORD' undeclared here (not in a function)
   96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                         ^~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/toolchain/common.h:168:26: note: in definition of macro '_DO_CONCAT'
  168 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/home/cfriedt/zephyrproject/zephyr/include/zephyr/device.h:96:33: note: in expansion of macro '_CONCAT'
   96 | #define DEVICE_NAME_GET(dev_id) _CONCAT(__device_, dev_id)
      |                                 ^~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/device.h:297:37: note: in expansion of macro 'DEVICE_NAME_GET'
  297 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_ID(node_id))
      |                                     ^~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/device.h:314:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  314 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:424:9: note: in expansion of macro 'DEVICE_DT_GET'
  424 |         DEVICE_DT_GET(DT_MTD_FROM_FIXED_PARTITION(DT_NODELABEL(label)))
      |         ^~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:37:33: note: in expansion of macro 'FIXED_PARTITION_DEVICE'
   37 | #define TEST_AREA_DEVICE        FIXED_PARTITION_DEVICE(TEST_AREA)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:67:47: note: in expansion of macro 'TEST_AREA_DEVICE'
   67 | static const struct device *const flash_dev = TEST_AREA_DEVICE;
      |                                               ^~~~~~~~~~~~~~~~
In file included from <command-line>:
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c: In function 'flash_driver_before':
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:36: error: 'DT_N_NODELABEL_storage_partition_REG_IDX_0_VAL_ADDRESSU' undeclared (first use in this function)
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h:102:23: note: in definition of macro '__UINT32_C'
  102 | #define __UINT32_C(c) c ## U
      |                       ^
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:9: note: in expansion of macro 'DT_U32_C'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |         ^~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2415:9: note: in expansion of macro 'DT_CAT4'
 2415 |         DT_CAT4(node_id, _REG_IDX_, idx, _VAL_ADDRESS)
      |         ^~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:18: note: in expansion of macro 'DT_REG_ADDR_BY_IDX_RAW'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |                  ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2461:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2461 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:39: note: in expansion of macro 'DT_REG_ADDR'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                       ^~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:29: note: in expansion of macro 'DT_CAT'
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:51: note: in expansion of macro 'DT_NODELABEL'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                                   ^~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:34:33: note: in expansion of macro 'FIXED_PARTITION_OFFSET'
   34 | #define TEST_AREA_OFFSET        FIXED_PARTITION_OFFSET(TEST_AREA)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:89:56: note: in expansion of macro 'TEST_AREA_OFFSET'
   89 |                 flash_get_page_info_by_offs(flash_dev, TEST_AREA_OFFSET,
      |                                                        ^~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:36: note: each undeclared identifier is reported only once for each function it appears in
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h:102:23: note: in definition of macro '__UINT32_C'
  102 | #define __UINT32_C(c) c ## U
      |                       ^
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:9: note: in expansion of macro 'DT_U32_C'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |         ^~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2415:9: note: in expansion of macro 'DT_CAT4'
 2415 |         DT_CAT4(node_id, _REG_IDX_, idx, _VAL_ADDRESS)
      |         ^~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:18: note: in expansion of macro 'DT_REG_ADDR_BY_IDX_RAW'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |                  ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2461:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2461 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:39: note: in expansion of macro 'DT_REG_ADDR'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                       ^~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:29: note: in expansion of macro 'DT_CAT'
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:51: note: in expansion of macro 'DT_NODELABEL'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                                   ^~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:34:33: note: in expansion of macro 'FIXED_PARTITION_OFFSET'
   34 | #define TEST_AREA_OFFSET        FIXED_PARTITION_OFFSET(TEST_AREA)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:89:56: note: in expansion of macro 'TEST_AREA_OFFSET'
   89 |                 flash_get_page_info_by_offs(flash_dev, TEST_AREA_OFFSET,
      |                                                        ^~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:36: error: 'DT_N_NODELABEL_storage_partition_REG_IDX_0_VAL_SIZEU' undeclared (first use in this function)
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h:102:23: note: in definition of macro '__UINT32_C'
  102 | #define __UINT32_C(c) c ## U
      |                       ^
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2452:9: note: in expansion of macro 'DT_U32_C'
 2452 |         DT_U32_C(DT_CAT4(node_id, _REG_IDX_, idx, _VAL_SIZE))
      |         ^~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2452:18: note: in expansion of macro 'DT_CAT4'
 2452 |         DT_U32_C(DT_CAT4(node_id, _REG_IDX_, idx, _VAL_SIZE))
      |                  ^~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2482:30: note: in expansion of macro 'DT_REG_SIZE_BY_IDX'
 2482 | #define DT_REG_SIZE(node_id) DT_REG_SIZE_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:395:37: note: in expansion of macro 'DT_REG_SIZE'
  395 | #define FIXED_PARTITION_SIZE(label) DT_REG_SIZE(DT_NODELABEL(label))
      |                                     ^~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:29: note: in expansion of macro 'DT_CAT'
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:395:49: note: in expansion of macro 'DT_NODELABEL'
  395 | #define FIXED_PARTITION_SIZE(label) DT_REG_SIZE(DT_NODELABEL(label))
      |                                                 ^~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:35:33: note: in expansion of macro 'FIXED_PARTITION_SIZE'
   35 | #define TEST_AREA_SIZE          FIXED_PARTITION_SIZE(TEST_AREA)
      |                                 ^~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:36:53: note: in expansion of macro 'TEST_AREA_SIZE'
   36 | #define TEST_AREA_MAX           (TEST_AREA_OFFSET + TEST_AREA_SIZE)
      |                                                     ^~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:96:35: note: in expansion of macro 'TEST_AREA_MAX'
   96 |                 page_info.size = (TEST_AREA_MAX - TEST_AREA_OFFSET) / 2;
      |                                   ^~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c: In function 'flash_driver_test_read_unaligned_address':
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:36: error: 'DT_N_NODELABEL_storage_partition_REG_IDX_0_VAL_ADDRESSU' undeclared (first use in this function)
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h:102:23: note: in definition of macro '__UINT32_C'
  102 | #define __UINT32_C(c) c ## U
      |                       ^
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:9: note: in expansion of macro 'DT_U32_C'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |         ^~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2415:9: note: in expansion of macro 'DT_CAT4'
 2415 |         DT_CAT4(node_id, _REG_IDX_, idx, _VAL_ADDRESS)
      |         ^~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:18: note: in expansion of macro 'DT_REG_ADDR_BY_IDX_RAW'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |                  ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2461:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2461 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:39: note: in expansion of macro 'DT_REG_ADDR'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                       ^~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:29: note: in expansion of macro 'DT_CAT'
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:51: note: in expansion of macro 'DT_NODELABEL'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                                   ^~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:34:33: note: in expansion of macro 'FIXED_PARTITION_OFFSET'
   34 | #define TEST_AREA_OFFSET        FIXED_PARTITION_OFFSET(TEST_AREA)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:164:25: note: in expansion of macro 'TEST_AREA_OFFSET'
  164 |                 start = TEST_AREA_OFFSET;
      |                         ^~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c: In function 'flash_driver_test_flash_fill':
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:36: error: 'DT_N_NODELABEL_storage_partition_REG_IDX_0_VAL_ADDRESSU' undeclared (first use in this function)
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h:102:23: note: in definition of macro '__UINT32_C'
  102 | #define __UINT32_C(c) c ## U
      |                       ^
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:9: note: in expansion of macro 'DT_U32_C'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |         ^~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2415:9: note: in expansion of macro 'DT_CAT4'
 2415 |         DT_CAT4(node_id, _REG_IDX_, idx, _VAL_ADDRESS)
      |         ^~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:18: note: in expansion of macro 'DT_REG_ADDR_BY_IDX_RAW'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |                  ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2461:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2461 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:39: note: in expansion of macro 'DT_REG_ADDR'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                       ^~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:29: note: in expansion of macro 'DT_CAT'
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:51: note: in expansion of macro 'DT_NODELABEL'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                                   ^~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:34:33: note: in expansion of macro 'FIXED_PARTITION_OFFSET'
   34 | #define TEST_AREA_OFFSET        FIXED_PARTITION_OFFSET(TEST_AREA)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:232:36: note: in expansion of macro 'TEST_AREA_OFFSET'
  232 |         rc = flash_read(flash_dev, TEST_AREA_OFFSET,
      |                                    ^~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c: In function 'flash_driver_test_flash_flatten':
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:36: error: 'DT_N_NODELABEL_storage_partition_REG_IDX_0_VAL_ADDRESSU' undeclared (first use in this function)
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h:102:23: note: in definition of macro '__UINT32_C'
  102 | #define __UINT32_C(c) c ## U
      |                       ^
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:9: note: in expansion of macro 'DT_U32_C'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |         ^~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2415:9: note: in expansion of macro 'DT_CAT4'
 2415 |         DT_CAT4(node_id, _REG_IDX_, idx, _VAL_ADDRESS)
      |         ^~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:18: note: in expansion of macro 'DT_REG_ADDR_BY_IDX_RAW'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |                  ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2461:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2461 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:39: note: in expansion of macro 'DT_REG_ADDR'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                       ^~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:29: note: in expansion of macro 'DT_CAT'
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:51: note: in expansion of macro 'DT_NODELABEL'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                                   ^~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:34:33: note: in expansion of macro 'FIXED_PARTITION_OFFSET'
   34 | #define TEST_AREA_OFFSET        FIXED_PARTITION_OFFSET(TEST_AREA)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:264:36: note: in expansion of macro 'TEST_AREA_OFFSET'
  264 |         rc = flash_read(flash_dev, TEST_AREA_OFFSET,
      |                                    ^~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c: In function 'flash_driver_test_flash_page_layout':
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:36: error: 'DT_N_NODELABEL_storage_partition_REG_IDX_0_VAL_ADDRESSU' undeclared (first use in this function)
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h:102:23: note: in definition of macro '__UINT32_C'
  102 | #define __UINT32_C(c) c ## U
      |                       ^
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:9: note: in expansion of macro 'DT_U32_C'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |         ^~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2415:9: note: in expansion of macro 'DT_CAT4'
 2415 |         DT_CAT4(node_id, _REG_IDX_, idx, _VAL_ADDRESS)
      |         ^~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:18: note: in expansion of macro 'DT_REG_ADDR_BY_IDX_RAW'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |                  ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2461:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2461 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:39: note: in expansion of macro 'DT_REG_ADDR'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                       ^~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:29: note: in expansion of macro 'DT_CAT'
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:51: note: in expansion of macro 'DT_NODELABEL'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                                   ^~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:34:33: note: in expansion of macro 'FIXED_PARTITION_OFFSET'
   34 | #define TEST_AREA_OFFSET        FIXED_PARTITION_OFFSET(TEST_AREA)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:375:53: note: in expansion of macro 'TEST_AREA_OFFSET'
  375 |         rc = flash_get_page_info_by_offs(flash_dev, TEST_AREA_OFFSET, &page_info_off);
      |                                                     ^~~~~~~~~~~~~~~~
In file included from /home/cfriedt/zephyrproject/zephyr/subsys/testsuite/ztest/include/zephyr/ztest.h:42,
                 from /home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:8:
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c: In function 'test_flash_copy_inner':
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:36: error: 'DT_N_NODELABEL_storage_partition_REG_IDX_0_VAL_ADDRESSU' undeclared (first use in this function)
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                                    ^~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/subsys/testsuite/ztest/include/zephyr/ztest_assert.h:180:35: note: in definition of macro '_zassert_base'
  180 |                         z_zassert(cond, _msg ? ("(" default_msg ")") : (default_msg), __FILE__,    \
      |                                   ^~~~
/home/cfriedt/zephyrproject/zephyr/subsys/testsuite/ztest/include/zephyr/ztest_assert.h:194:9: note: in expansion of macro '_zassert_va'
  194 |         _zassert_va(cond, default_msg, COND_CODE_1(__VA_OPT__(1), (__VA_ARGS__), (NULL)))
      |         ^~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/subsys/testsuite/ztest/include/zephyr/ztest_assert.h:289:31: note: in expansion of macro 'zassert'
  289 | #define zassert_ok(cond, ...) zassert(!(cond), #cond " is non-zero", ##__VA_ARGS__)
      |                               ^~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:436:17: note: in expansion of macro 'zassert_ok'
  436 |                 zassert_ok(flash_read(flash_dev, TEST_AREA_OFFSET, expected, copy_size));
      |                 ^~~~~~~~~~
/opt/zephyr/zephyr-sdk-0.17.2/arm-zephyr-eabi/picolibc/include/stdint.h:420:21: note: in expansion of macro '__UINT32_C'
  420 | #define UINT32_C(x) __UINT32_C(x)
      |                     ^~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:9: note: in expansion of macro 'DT_U32_C'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |         ^~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2415:9: note: in expansion of macro 'DT_CAT4'
 2415 |         DT_CAT4(node_id, _REG_IDX_, idx, _VAL_ADDRESS)
      |         ^~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2438:18: note: in expansion of macro 'DT_REG_ADDR_BY_IDX_RAW'
 2438 |         DT_U32_C(DT_REG_ADDR_BY_IDX_RAW(node_id, idx))
      |                  ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:2461:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2461 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:39: note: in expansion of macro 'DT_REG_ADDR'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                       ^~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/devicetree.h:196:29: note: in expansion of macro 'DT_CAT'
  196 | #define DT_NODELABEL(label) DT_CAT(DT_N_NODELABEL_, label)
      |                             ^~~~~~
/home/cfriedt/zephyrproject/zephyr/include/zephyr/storage/flash_map.h:377:51: note: in expansion of macro 'DT_NODELABEL'
  377 | #define FIXED_PARTITION_OFFSET(label) DT_REG_ADDR(DT_NODELABEL(label))
      |                                                   ^~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:34:33: note: in expansion of macro 'FIXED_PARTITION_OFFSET'
   34 | #define TEST_AREA_OFFSET        FIXED_PARTITION_OFFSET(TEST_AREA)
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
/home/cfriedt/zephyrproject/zephyr/tests/drivers/flash/common/src/main.c:436:50: note: in expansion of macro 'TEST_AREA_OFFSET'
  436 |                 zassert_ok(flash_read(flash_dev, TEST_AREA_OFFSET, expected, copy_size));
      |                                                  ^~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /opt/zephyr/venv/20250910/bin/cmake --build /home/cfriedt/zephyrproject/zephyr/build
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- **OS:** Linux
- **Toolchain:** Zephyr SDK
- **Commit SHA:** 7fc8051b0f4b4299034fb110f821adeba3445368

### Additional Context

Some failing CI runs. These have existed for several weeks / months.

https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729502
https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729501
https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729559
https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729570
https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729577
https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729651
https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729716
https://github.com/zephyrproject-rtos/zephyr/actions/runs/18068117603/job/51413729715

--- a/tests/drivers/flash/common/Kconfig
+++ b/tests/drivers/flash/common/Kconfig
@@ -16,4 +16,12 @@ config TEST_DRIVER_FLASH_SIZE
 	  support the get_size() API, leave this set as -1 to skip the test.
 	  For the STM32 devices, the flash size is directly given by the soc DTSI.
 
+config TEST_FORCE_STORAGE_PARTITION
+	bool "Force testing the storage partition area"
+	default n
+	help
+	  This test prioritizes testing spi nor flash over
+	  the storage area by default. Set this config to priorize
+	  testing the storage partition instead.
+
 source "Kconfig.zephyr"

--- a/tests/drivers/flash/common/src/main.c
+++ b/tests/drivers/flash/common/src/main.c
@@ -10,7 +10,9 @@
 #include <zephyr/devicetree.h>
 #include <zephyr/storage/flash_map.h>
 
-#if defined(CONFIG_NORDIC_QSPI_NOR)
+#if defined(CONFIG_TEST_FORCE_STORAGE_PARTITION)
+#define TEST_AREA	storage_partition
+#elif defined(CONFIG_NORDIC_QSPI_NOR)
 #define TEST_AREA_DEV_NODE	DT_INST(0, nordic_qspi_nor)
 #elif defined(SOC_SERIES_STM32N6X)
 #define TEST_AREA_DEV_NODE	DT_INST(0, st_stm32_xspi_nor)

--- a/tests/drivers/flash/common/testcase.yaml
+++ b/tests/drivers/flash/common/testcase.yaml
@@ -155,12 +155,12 @@ tests:
     platform_allow:
       - nrf52840dk/nrf52840
       - nrf5340dk/nrf5340/cpuapp
-  drivers.flash.common.disable_spi_nor:
-    filter: dt_compat_enabled("soc-nv-flash") and dt_compat_enabled("jedec,spi-nor")
+  drivers.flash.common.test_storage_partition:
+    filter: dt_label_with_parent_compat_enabled("storage_partition", "fixed-partitions")
     platform_exclude:
       - beagleconnect_freedom/cc1352p7
     extra_args:
-      - CONFIG_SPI_NOR=n
+      - CONFIG_TEST_FORCE_STORAGE_PARTITION=y
   drivers.flash.common.ra_ospi_b_nor:
     platform_allow:
       - ek_ra8m1

